### PR TITLE
Frontend 

### PR DESCRIPTION
**Pull request by samBAM089**
- login
- first page (items are not loading from backend yet)

### CODE REVIEW

File: frontend/.prettierrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (jamarob) [inline frontend/.prettierrc:1]: ❤️

File: frontend/public/index.html [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
-    <!--
+    <head>
+        <meta charset="utf-8" />
+        <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+        <meta name="theme-color" content="#000000" />
+        <meta
+            name="description"
+            content="Web site created using create-react-app"
```
Comment (jamarob) [inline frontend/public/index.html:10]: Add your own description!

File: frontend/src/App.test.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { render, screen } from '@testing-library/react';
 import App from './App';
 
 test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+    render(<App />);
+    const linkElement = screen.getByText(/learn react/i);
```
Comment (jamarob) [inline frontend/src/App.test.js:6]: This test will fail ;)

File: frontend/src/components/ButtonBig.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import styled from 'styled-components/macro';
+
+export default function ButtonBig() {
+    return <Button> LET'S PLAY </Button>;
+}
+
+const Button = styled.button`
+    border-radius: 50px;
```
Comment (jamarob) [inline frontend/src/components/ButtonBig.js:8]: this is a huge border radius, for completely round corners use `border-radius: 50%`

File: frontend/src/components/ButtonBig.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    padding: 10px 50px;
+    margin: 8px;
+    background: #c8a1a2;
+    color: white;
+    border: none;
```
Comment (jamarob) [inline frontend/src/components/ButtonBig.js:13]: Think about adding a little shadow to make the button 'pop'

File: frontend/src/components/Footer.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import styled from 'styled-components/macro';
+
+export default function Footer() {
+    return (
+        <Wrapper>
+            <h5>powered by</h5>
```
Comment (jamarob) [inline frontend/src/components/Footer.js:6]: Do you have `h1` to `h4` on the page? Don't use headings for styling, better respect html semantics and use the correct level of heading and style it how you need it.

File: frontend/src/components/Footer.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+            <a
+                href="https://www.instagram.com/brettspielfieber/?hl=de}"
+                target="blank"
```
Comment (jamarob) [inline frontend/src/components/Footer.js:9]:
```suggestion
                target="_blank"
```

File: frontend/src/components/Footer.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+            >
+                <img src="./images/bf-logo.png" alt=" logo" />
```
Comment (jamarob) [inline frontend/src/components/Footer.js:11]: In the alt attribute say which logo it is.

File: frontend/src/components/Footer.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        </Wrapper>
+    );
+}
+
+const Wrapper = styled.section`
+    display: grid;
+    grid-template-columns: auto auto;
+    background: var(--primary);
+    margin: 0;
+    padding: 0;
```
Comment (jamarob) [inline frontend/src/components/Footer.js:22]:
```suggestion
```

default for sections..

File: frontend/src/components/Header.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
+import styled from 'styled-components/macro';
+
+export default function Header() {
+    return (
+        <Wrapper>
+            <div>
```
Comment (jamarob) [inline frontend/src/components/Header.js:6]: Why the `div`? I think this can be deleted

File: frontend/src/components/PageLayout.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+
+export default styled.div`
+    height: 100vh;
+    display: grid;
+    grid-template-rows: auto 1fr auto auto;
+
+    main {
+        background: var(--primary);
+        padding: 0 16px;
+        overflow-y: hidden;
```
Comment (jamarob) [inline frontend/src/components/PageLayout.js:11]: Are you sure? I think you want to scroll on overflow, no?

File: frontend/src/components/SessionBoard.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    return (
+        <>
+            <Wrapper>
+                {sessions.map((session) => (
+                    <li key={session.id}>
+                        <SessionTile session={session} />
+                    </li>
+                ))}
+            </Wrapper>
+        </>
```
Comment (jamarob) [inline frontend/src/components/SessionBoard.js:14]:
```suggestion
            <Wrapper>
                {sessions.map((session) => (
                    <li key={session.id}>
                        <SessionTile session={session} />
                    </li>
                ))}
            </Wrapper>
```

File: frontend/src/components/SessionBoard.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    );
+}
+
+const Wrapper = styled.ul`
+    display: grid;
+    justify-content: center;
+    height: 100%;
+    background: none;
+    overflow-y: scroll;
```
Comment (jamarob) [inline frontend/src/components/SessionBoard.js:23]: Ah, refering to my comment in `PageLayout`, making the main scroll would be more flexible

File: frontend/src/components/SessionBoard.js [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+            </Wrapper>
+        </>
+    );
+}
+
+const Wrapper = styled.ul`
+    display: grid;
+    justify-content: center;
+    height: 100%;
+    background: none;
```
Comment (jamarob) [inline frontend/src/components/SessionBoard.js:22]: I think you don't need this

File: frontend/src/components/SessionTile.js [hunk 15]
```diff
@@ -0,0 +1,50 @@
+import styled from 'styled-components/macro';
+
+export default function SessionTile({ session }) {
+    return (
+        <Wrapper>
+            <span>
+                <img src={session.thumbnail} alt="" />
+            </span>
```
Comment (jamarob) [inline frontend/src/components/SessionTile.js:8]: Why the span? I think this can be safely deleted

File: frontend/src/components/SessionTile.js [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+            <div>
+                <span>{session.startDate}</span>
```
Comment (jamarob) [inline frontend/src/components/SessionTile.js:10]:
```suggestion
                <time>{session.startDate}</time>
```

File: frontend/src/components/SessionTile.js [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+                <br />
+                <span>
+                    <h2>{session.gameName}</h2>
+                </span>
```
Comment (jamarob) [inline frontend/src/components/SessionTile.js:14]:
```suggestion
                    <h2>{session.gameName}</h2>
```

File: frontend/src/components/SessionTile.js [hunk 18]
```diff
@@ -0,0 +1,50 @@
+import styled from 'styled-components/macro';
+
+export default function SessionTile({ session }) {
+    return (
+        <Wrapper>
```
Comment (jamarob) [inline frontend/src/components/SessionTile.js:5]: Think about a different layout that doesn't need the `<br />`, also there is some unecessary nesting I think

File: frontend/src/components/SessionTile.js [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+                <br />
+
+                <span>Winner: {session.winner}</span>
+            </div>
+        </Wrapper>
+    );
+}
+
+const Wrapper = styled.section`
+    background: none;
```
Comment (jamarob) [inline frontend/src/components/SessionTile.js:24]: not neeeded

File: frontend/src/components/SessionTile.js [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+    border: 1px solid white;
+    list-style: none;
```
Comment (jamarob) [inline frontend/src/components/SessionTile.js:26]: not a list

File: frontend/src/pages/Login.js [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+import { loginUser } from '../services/loginService';
+import { Redirect } from 'react-router-dom';
+
+export default function Login({ setJwtToken, jwtToken }) {
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        if (!username && !password) {
```
Comment (jamarob) [inline frontend/src/pages/Login.js:14]:
```suggestion
        if (!username || !password) {
```

File: frontend/src/pages/Login.js [hunk 22]
```diff
@@ -0,0 +1,87 @@
...
+            return;
+        }
+        loginUser(username, password).then(setJwtToken);
```
Comment (jamarob) [inline frontend/src/pages/Login.js:17]: Error handling!

File: frontend/src/pages/Login.js [hunk 23]
```diff
@@ -0,0 +1,87 @@
...
+    };
+
+    if (jwtToken) {
+        return <Redirect to={'/'} />;
+    }
+
+    return (
+        <PageLayout>
+            <main>
+                <Wrapper>
```
Comment (jamarob) [inline frontend/src/pages/Login.js:29]: I think the main is not needed if you base `Wrapper` on a main tag. And even that may be not needed, you could restyle the PageLayout?

File: frontend/src/pages/StartSession.js [hunk 24]
```diff
@@ -0,0 +1,5 @@
+import SessionTile from '../components/SessionTile';
+
+export default function StartSession() {
+    return 'hallo';
```
Comment (jamarob) [inline frontend/src/pages/StartSession.js:4]: hi 👋

File: frontend/src/pages/Overview.js [hunk 25]
```diff
@@ -0,0 +1,66 @@
+import SessionBoard from '../components/SessionBoard';
+import PageLayout from '../components/PageLayout';
+import Header from '../components/Header';
+import Footer from '../components/Footer';
+import ButtonTabBottom from '../components/ButtonTabBottom';
+import { Redirect } from 'react-router-dom';
+
+export default function Overview({ jwtToken }) {
+    const sessions = [
```
Comment (jamarob) [inline frontend/src/pages/Overview.js:9]: Maybe extract to an extra file, even if it is only temporary it hinders readability :)

**Review by jamarob — COMMENTED** [review]
Well played 🎲  Please take some tie to clean up your css there are some rules that are not needed, also try to keep nesting to a minimum, you have some divs and spans that can be deleted.

File: frontend/src/App.test.js (hunk 3 above)
Comment (samBAM089) [reply frontend/src/App.test.js:6]: i do hope it makes sense to do FR testing at the very end.... ^^'

File: frontend/src/components/ButtonBig.js (hunk 4 above)
Comment (samBAM089) [reply frontend/src/components/ButtonBig.js:8]: check ^^

File: frontend/public/index.html (hunk 2 above)
Comment (samBAM089) [reply frontend/public/index.html:10]: done ^^

File: frontend/src/components/ButtonBig.js (hunk 5 above)
Comment (samBAM089) [reply frontend/src/components/ButtonBig.js:13]: added ^^

File: frontend/src/pages/Login.js (hunk 21 above)
Comment (samBAM089) [reply frontend/src/pages/Login.js:14]: oh, makes sense ^^'

File: frontend/src/pages/Login.js (hunk 22 above)
Comment (samBAM089) [reply frontend/src/pages/Login.js:17]: done! while I might need some pointers for transitions later ^^'

File: frontend/src/pages/Login.js (hunk 23 above)
Comment (samBAM089) [reply frontend/src/pages/Login.js:29]: I tried to take it out and change my PageLayout class, but it results in wrong rendering on other pages...

File: frontend/src/pages/Overview.js (hunk 25 above)
Comment (samBAM089) [reply frontend/src/pages/Overview.js:9]: it will be gone as soon as I get the real sessions from the backend ^^'

**Comment by samBAM089** [comment]
changes add ^^

File: frontend/src/App.test.js (hunk 3 above)
Comment (jamarob) [reply frontend/src/App.test.js:6]: 👍 a failing test will break your ci pipeline, so i suggest you delete this file

File: frontend/public/index.html (hunk 2 above)
Comment (jamarob) [reply frontend/public/index.html:10]: I said description, but that's the name of the property. Replace content with your description.

**Review by jamarob — APPROVED** [review]
Gj!

**Pull request merged**